### ISSUE TITLE
pull access denied

### ISSUE THREAD

**Issue by dpositive** [toby-coleman/coffee-roaster#2]
I've made everything as was mentioned in readme file but have build fail with the following overview:

[Info]     Starting build for roaster, user gh_dpositive
[Info]     Dashboard link: https://dashboard.balena-cloud.com/apps/1609230/devices
[Info]     Building on arm01
[Info]     Pulling previous images for caching purposes...
[Success]  Successfully pulled cache images
[Info]     Starting to build nodered
[Info]     Starting to build web
[Info]     Starting to build redis
[Success]  Successfully built service redis
[Info]     Uploading images
[Success]  Successfully uploaded images
[Error]    Some services failed to build:
[Error]      Service: nodered
[Error]        Error: pull access denied for resin/raspberrypi3-64-alpine-node, repository does not exist or may require 'docker login'
[Error]      Service: web
[Error]        Error: The command '/bin/sh -c pip install numpy --extra-index-url https://www.piwheels.org/simple' returned a non-zero code: 1
[Info]     Built on arm01
[Error]    Not deploying release.

Do you have any idea how to solve it?

**Comment by toby-coleman** [comment]
Looks like Balena have renamed their Docker container since I last built the image. The new name is [balenalib/raspberrypi3-alpine-node](https://hub.docker.com/r/balenalib/raspberrypi3-alpine) instead of resin/raspberrypi3-64-alpine-node.  I'll test the changes and let you know when it is ready to try again.

**Comment by dpositive** [comment]
Thanks for quick response. I will wait

**Comment by toby-coleman** [comment]
Just tested again on Balena and it builds with the latest commit that I just pushed

**Comment by dpositive** [comment]
Ok, but now there is a problem with the web:

Step 1/9 : FROM balenalib/raspberrypi3-64-python:3.7
 ---> 8312331d9457
Step 2/9 : RUN mkdir -p /usr/src/app/
Using cache
 ---> 4dd89ea8e706
Step 3/9 : COPY app/requirements.txt /usr/src/app/requirements.txt
Using cache
 ---> 41cdce0791cb
Step 4/9 : WORKDIR /usr/src/app
Using cache
 ---> a5277bc2fba1
Step 5/9 : RUN apt-get update -y &&   apt-get -y install libatlas-base-dev &&   apt-get clean && rm -rf /var/lib/apt/lists/*
Using cache
 ---> fdd3d29ee6a3
Step 6/9 : RUN pip install numpy --extra-index-url https://www.piwheels.org/simple
 ---> Running in d3a18ffc138c
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting numpy
  Downloading https://files.pythonhosted.org/packages/40/de/0ea5092b8bfd2e3aa6fdbb2e499a9f9adf810992884d414defc1573dca3f/numpy-1.18.1.zip (5.4MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /usr/local/bin/python3.7 /usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmptq6uov0e
         cwd: /tmp/pip-install-wudi_4ni/numpy
    Complete output (223 lines):
    Processing numpy/random/_bounded_integers.pxd.in
    Processing numpy/random/_generator.pyx
    Processing numpy/random/_common.pyx
    Processing numpy/random/_mt19937.pyx
    Processing numpy/random/mtrand.pyx
    Processing numpy/random/_bit_generator.pyx
    Processing numpy/random/_philox.pyx
    Processing numpy/random/_pcg64.pyx
    Processing numpy/random/_sfc64.pyx
    Processing numpy/random/_bounded_integers.pyx.in
    Cythonizing sources
    blas_opt_info:
    blas_mkl_info:
    customize UnixCCompiler
      libraries mkl_rt not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    blis_info:
      libraries blis not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    openblas_info:
      libraries openblas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    atlas_3_10_blas_threads_info:
    Setting PTATLAS=ATLAS
      libraries tatlas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    atlas_3_10_blas_info:
      libraries satlas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    atlas_blas_threads_info:
    Setting PTATLAS=ATLAS
      libraries ptf77blas,ptcblas,atlas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    atlas_blas_info:
      libraries f77blas,cblas,atlas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    accelerate_info:
      NOT AVAILABLE
    blas_info:
      libraries blas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    blas_src_info:
      NOT AVAILABLE
      NOT AVAILABLE
    /bin/sh: 1: svnversion: not found
    non-existing path in 'numpy/distutils': 'site.cfg'
    lapack_opt_info:
    lapack_mkl_info:
      libraries mkl_rt not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    openblas_lapack_info:
      libraries openblas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    openblas_clapack_info:
      libraries openblas,lapack not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    flame_info:
      libraries flame not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    atlas_3_10_threads_info:
    Setting PTATLAS=ATLAS
      libraries lapack_atlas not found in /usr/local/lib
      libraries tatlas,tatlas not found in /usr/local/lib
      libraries lapack_atlas not found in /usr/lib
      libraries tatlas,tatlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>
      NOT AVAILABLE
    atlas_3_10_info:
      libraries lapack_atlas not found in /usr/local/lib
      libraries satlas,satlas not found in /usr/local/lib
      libraries lapack_atlas not found in /usr/lib
      libraries satlas,satlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_3_10_info'>
      NOT AVAILABLE
    atlas_threads_info:
    Setting PTATLAS=ATLAS
      libraries lapack_atlas not found in /usr/local/lib
      libraries ptf77blas,ptcblas,atlas not found in /usr/local/lib
      libraries lapack_atlas not found in /usr/lib
      libraries ptf77blas,ptcblas,atlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_threads_info'>
      NOT AVAILABLE
    atlas_info:
      libraries lapack_atlas not found in /usr/local/lib
      libraries f77blas,cblas,atlas not found in /usr/local/lib
      libraries lapack_atlas not found in /usr/lib
      libraries f77blas,cblas,atlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_info'>
      NOT AVAILABLE
    lapack_info:
      libraries lapack not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    lapack_src_info:
      NOT AVAILABLE
      NOT AVAILABLE
    running dist_info
    running build_src
    build_src
    building py_modules sources
    creating build
    creating build/src.linux-aarch64-3.7
    creating build/src.linux-aarch64-3.7/numpy
    creating build/src.linux-aarch64-3.7/numpy/distutils
    building library "npymath" sources
    Could not locate executable gfortran
    Could not locate executable f95
    Could not locate executable ifort
    Could not locate executable ifc
    Could not locate executable lf95
    Could not locate executable pgfortran
    Could not locate executable f90
    Could not locate executable f77
    Could not locate executable fort
    Could not locate executable efort
    Could not locate executable efc
    Could not locate executable g77
    Could not locate executable g95
    Could not locate executable pathf95
    Could not locate executable nagfor
    don't know how to compile Fortran code on platform 'posix'
    Running from numpy source directory.
    setup.py:461: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates
      run_build = parse_setuppy_commands()
    /tmp/pip-install-wudi_4ni/numpy/numpy/distutils/system_info.py:1896: UserWarning:
        Optimized (vendor) Blas libraries are not found.
        Falls back to netlib Blas library which has worse performance.
        A better performance should be easily gained by switching
        Blas library.
      if self._calc_info(blas):
    /tmp/pip-install-wudi_4ni/numpy/numpy/distutils/system_info.py:1896: UserWarning:
        Blas (http://www.netlib.org/blas/) libraries not found.
        Directories to search for the libraries can be specified in the
        numpy/distutils/site.cfg file (section [blas]) or by setting
        the BLAS environment variable.
      if self._calc_info(blas):
    /tmp/pip-install-wudi_4ni/numpy/numpy/distutils/system_info.py:1896: UserWarning:
        Blas (http://www.netlib.org/blas/) sources not found.
        Directories to search for the sources can be specified in the
        numpy/distutils/site.cfg file (section [blas_src]) or by setting
        the BLAS_SRC environment variable.
      if self._calc_info(blas):
    /tmp/pip-install-wudi_4ni/numpy/numpy/distutils/system_info.py:1730: UserWarning:
        Lapack (http://www.netlib.org/lapack/) libraries not found.
        Directories to search for the libraries can be specified in the
        numpy/distutils/site.cfg file (section [lapack]) or by setting
        the LAPACK environment variable.
      return getattr(self, '_calc_info_{}'.format(name))()
    /tmp/pip-install-wudi_4ni/numpy/numpy/distutils/system_info.py:1730: UserWarning:
        Lapack (http://www.netlib.org/lapack/) sources not found.
        Directories to search for the sources can be specified in the
        numpy/distutils/site.cfg file (section [lapack_src]) or by setting
        the LAPACK_SRC environment variable.
      return getattr(self, '_calc_info_{}'.format(name))()
    /usr/local/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'define_macros'
      warnings.warn(msg)
    Traceback (most recent call last):
      File "/usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 257, in <module>
        main()
      File "/usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 240, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 110, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
      File "/tmp/pip-build-env-v2y35prl/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 158, in prepare_metadata_for_build_wheel
        self.run_setup()
      File "/tmp/pip-build-env-v2y35prl/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 250, in run_setup
        self).run_setup(setup_script=setup_script)
      File "/tmp/pip-build-env-v2y35prl/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 143, in run_setup
        exec(compile(code, __file__, 'exec'), locals())
      File "setup.py", line 488, in <module>
        setup_package()
      File "setup.py", line 480, in setup_package
        setup(**metadata)
      File "/tmp/pip-install-wudi_4ni/numpy/numpy/distutils/core.py", line 171, in setup
        return old_setup(**new_attr)
      File "/tmp/pip-build-env-v2y35prl/overlay/lib/python3.7/site-packages/setuptools/__init__.py", line 144, in setup
        return distutils.core.setup(**attrs)
      File "/usr/local/lib/python3.7/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/local/lib/python3.7/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/usr/local/lib/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-build-env-v2y35prl/overlay/lib/python3.7/site-packages/setuptools/command/dist_info.py", line 31, in run
        egg_info.run()
      File "/tmp/pip-install-wudi_4ni/numpy/numpy/distutils/command/egg_info.py", line 26, in run
        self.run_command("build_src")
      File "/usr/local/lib/python3.7/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/local/lib/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-install-wudi_4ni/numpy/numpy/distutils/command/build_src.py", line 146, in run
        self.build_sources()
      File "/tmp/pip-install-wudi_4ni/numpy/numpy/distutils/command/build_src.py", line 157, in build_sources
        self.build_library_sources(*libname_info)
      File "/tmp/pip-install-wudi_4ni/numpy/numpy/distutils/command/build_src.py", line 290, in build_library_sources
        sources = self.generate_sources(sources, (lib_name, build_info))
      File "/tmp/pip-install-wudi_4ni/numpy/numpy/distutils/command/build_src.py", line 380, in generate_sources
        source = func(extension, build_dir)
      File "numpy/core/setup.py", line 661, in get_mathlib_info
        raise RuntimeError("Broken toolchain: cannot link a simple C program")
    RuntimeError: Broken toolchain: cannot link a simple C program
    ----------------------------------------

ERROR: Command errored out with exit status 1: /usr/local/bin/python3.7 /usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmptq6uov0e Check the logs for full command output.

WARNING: You are using pip version 19.3.1; however, version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

Removing intermediate container d3a18ffc138c
The command '/bin/sh -c pip install numpy --extra-index-url https://www.piwheels.org/simple' returned a non-zero code: 1

**Comment by toby-coleman** [comment]
Sorry about that. Can you try again please with the latest commit?

**Comment by dpositive** [comment]
Still have the same issue
```
Step 1/9 : FROM balenalib/raspberrypi3-64-python:3.7
 ---> 8312331d9457
Step 2/9 : RUN mkdir -p /usr/src/app/
Using cache
 ---> 4dd89ea8e706
Step 3/9 : COPY app/requirements.txt /usr/src/app/requirements.txt
Using cache
 ---> 1280692c5dae
Step 4/9 : WORKDIR /usr/src/app
Using cache
 ---> 72394af41bd4
Step 5/9 : RUN apt-get update -y &&   apt-get -y install libatlas3-base libgfortran5 &&   apt-get clean && rm -rf /var/lib/apt/lists/*
Using cache
 ---> de3a6fed297c
Step 6/9 : RUN pip install numpy --extra-index-url https://www.piwheels.org/simple
 ---> Running in aa54b7bcc53a
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting numpy
  Downloading https://files.pythonhosted.org/packages/40/de/0ea5092b8bfd2e3aa6fdbb2e499a9f9adf810992884d414defc1573dca3f/numpy-1.18.1.zip (5.4MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /usr/local/bin/python3.7 /usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpswjooiqv
         cwd: /tmp/pip-install-hjs3qdyo/numpy
    Complete output (223 lines):
    Processing numpy/random/_bounded_integers.pxd.in
    Processing numpy/random/_mt19937.pyx
    Processing numpy/random/mtrand.pyx
    Processing numpy/random/_pcg64.pyx
    Processing numpy/random/_bit_generator.pyx
    Processing numpy/random/_generator.pyx
    Processing numpy/random/_philox.pyx
    Processing numpy/random/_sfc64.pyx
    Processing numpy/random/_bounded_integers.pyx.in
    Processing numpy/random/_common.pyx
    Cythonizing sources
    blas_opt_info:
    blas_mkl_info:
    customize UnixCCompiler
      libraries mkl_rt not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    blis_info:
      libraries blis not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    openblas_info:
      libraries openblas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    atlas_3_10_blas_threads_info:
    Setting PTATLAS=ATLAS
      libraries tatlas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    atlas_3_10_blas_info:
      libraries satlas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    atlas_blas_threads_info:
    Setting PTATLAS=ATLAS
      libraries ptf77blas,ptcblas,atlas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    atlas_blas_info:
      libraries f77blas,cblas,atlas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    accelerate_info:
      NOT AVAILABLE
    blas_info:
      libraries blas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    blas_src_info:
      NOT AVAILABLE
      NOT AVAILABLE
    /bin/sh: 1: svnversion: not found
    non-existing path in 'numpy/distutils': 'site.cfg'
    lapack_opt_info:
    lapack_mkl_info:
      libraries mkl_rt not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    openblas_lapack_info:
      libraries openblas not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    openblas_clapack_info:
      libraries openblas,lapack not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    flame_info:
      libraries flame not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    atlas_3_10_threads_info:
    Setting PTATLAS=ATLAS
      libraries lapack_atlas not found in /usr/local/lib
      libraries tatlas,tatlas not found in /usr/local/lib
      libraries lapack_atlas not found in /usr/lib
      libraries tatlas,tatlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>
      NOT AVAILABLE
    atlas_3_10_info:
      libraries lapack_atlas not found in /usr/local/lib
      libraries satlas,satlas not found in /usr/local/lib
      libraries lapack_atlas not found in /usr/lib
      libraries satlas,satlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_3_10_info'>
      NOT AVAILABLE
    atlas_threads_info:
    Setting PTATLAS=ATLAS
      libraries lapack_atlas not found in /usr/local/lib
      libraries ptf77blas,ptcblas,atlas not found in /usr/local/lib
      libraries lapack_atlas not found in /usr/lib
      libraries ptf77blas,ptcblas,atlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_threads_info'>
      NOT AVAILABLE
    atlas_info:
      libraries lapack_atlas not found in /usr/local/lib
      libraries f77blas,cblas,atlas not found in /usr/local/lib
      libraries lapack_atlas not found in /usr/lib
      libraries f77blas,cblas,atlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_info'>
      NOT AVAILABLE
    lapack_info:
      libraries lapack not found in ['/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    lapack_src_info:
      NOT AVAILABLE
      NOT AVAILABLE
    running dist_info
    running build_src
    build_src
    building py_modules sources
    creating build
    creating build/src.linux-aarch64-3.7
    creating build/src.linux-aarch64-3.7/numpy
    creating build/src.linux-aarch64-3.7/numpy/distutils
    building library "npymath" sources
    Could not locate executable gfortran
    Could not locate executable f95
    Could not locate executable ifort
    Could not locate executable ifc
    Could not locate executable lf95
    Could not locate executable pgfortran
    Could not locate executable f90
    Could not locate executable f77
    Could not locate executable fort
    Could not locate executable efort
    Could not locate executable efc
    Could not locate executable g77
    Could not locate executable g95
    Could not locate executable pathf95
    Could not locate executable nagfor
    don't know how to compile Fortran code on platform 'posix'
    Running from numpy source directory.
    setup.py:461: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates
      run_build = parse_setuppy_commands()
    /tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/system_info.py:1896: UserWarning:
        Optimized (vendor) Blas libraries are not found.
        Falls back to netlib Blas library which has worse performance.
        A better performance should be easily gained by switching
        Blas library.
      if self._calc_info(blas):
    /tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/system_info.py:1896: UserWarning:
        Blas (http://www.netlib.org/blas/) libraries not found.
        Directories to search for the libraries can be specified in the
        numpy/distutils/site.cfg file (section [blas]) or by setting
        the BLAS environment variable.
      if self._calc_info(blas):
    /tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/system_info.py:1896: UserWarning:
        Blas (http://www.netlib.org/blas/) sources not found.
        Directories to search for the sources can be specified in the
        numpy/distutils/site.cfg file (section [blas_src]) or by setting
        the BLAS_SRC environment variable.
      if self._calc_info(blas):
    /tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/system_info.py:1730: UserWarning:
        Lapack (http://www.netlib.org/lapack/) libraries not found.
        Directories to search for the libraries can be specified in the
        numpy/distutils/site.cfg file (section [lapack]) or by setting
        the LAPACK environment variable.
      return getattr(self, '_calc_info_{}'.format(name))()
    /tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/system_info.py:1730: UserWarning:
        Lapack (http://www.netlib.org/lapack/) sources not found.
        Directories to search for the sources can be specified in the
        numpy/distutils/site.cfg file (section [lapack_src]) or by setting
        the LAPACK_SRC environment variable.
      return getattr(self, '_calc_info_{}'.format(name))()
    /usr/local/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'define_macros'
      warnings.warn(msg)
    Traceback (most recent call last):
      File "/usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 257, in <module>
        main()
      File "/usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 240, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py", line 110, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
      File "/tmp/pip-build-env-7ombmp1e/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 158, in prepare_metadata_for_build_wheel
        self.run_setup()
      File "/tmp/pip-build-env-7ombmp1e/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 250, in run_setup
        self).run_setup(setup_script=setup_script)
      File "/tmp/pip-build-env-7ombmp1e/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 143, in run_setup
        exec(compile(code, __file__, 'exec'), locals())
      File "setup.py", line 488, in <module>
        setup_package()
      File "setup.py", line 480, in setup_package
        setup(**metadata)
      File "/tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/core.py", line 171, in setup
        return old_setup(**new_attr)
      File "/tmp/pip-build-env-7ombmp1e/overlay/lib/python3.7/site-packages/setuptools/__init__.py", line 144, in setup
        return distutils.core.setup(**attrs)
      File "/usr/local/lib/python3.7/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/usr/local/lib/python3.7/distutils/dist.py", line 966, in run_commands
        self.run_command(cmd)
      File "/usr/local/lib/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-build-env-7ombmp1e/overlay/lib/python3.7/site-packages/setuptools/command/dist_info.py", line 31, in run
        egg_info.run()
      File "/tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/command/egg_info.py", line 26, in run
        self.run_command("build_src")
      File "/usr/local/lib/python3.7/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/usr/local/lib/python3.7/distutils/dist.py", line 985, in run_command
        cmd_obj.run()
      File "/tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/command/build_src.py", line 146, in run
        self.build_sources()
      File "/tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/command/build_src.py", line 157, in build_sources
        self.build_library_sources(*libname_info)
      File "/tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/command/build_src.py", line 290, in build_library_sources
        sources = self.generate_sources(sources, (lib_name, build_info))
      File "/tmp/pip-install-hjs3qdyo/numpy/numpy/distutils/command/build_src.py", line 380, in generate_sources
        source = func(extension, build_dir)
      File "numpy/core/setup.py", line 661, in get_mathlib_info
        raise RuntimeError("Broken toolchain: cannot link a simple C program")
    RuntimeError: Broken toolchain: cannot link a simple C program
    ----------------------------------------

ERROR: Command errored out with exit status 1: /usr/local/bin/python3.7 /usr/local/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpswjooiqv Check the logs for full command output.

WARNING: You are using pip version 19.3.1; however, version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

Removing intermediate container aa54b7bcc53a
The command '/bin/sh -c pip install numpy --extra-index-url https://www.piwheels.org/simple' returned a non-zero code: 1
```

**Comment by toby-coleman** [comment]
Hmm OK - I'm guessing the Docker base image has been changed, and my build is still running off a cached version.  Let me try again...

**Comment by toby-coleman** [comment]
Right - hopefully got it this time. The web container built entirely (I checked that it refreshed completely the base container)
[Info]     ┌─────────┬────────────┬───────────────────────┐
[Info]     │ Service │ Image Size │ Build Time            │
[Info]     ├─────────┼────────────┼───────────────────────┤
[Info]     │ redis   │ 68.80 MB   │ 1 second              │
[Info]     ├─────────┼────────────┼───────────────────────┤
[Info]     │ nodered │ 334.66 MB  │ 1 second              │
[Info]     ├─────────┼────────────┼───────────────────────┤
[Info]     │ web     │ 1.07 GB    │ 5 minutes, 55 seconds │
[Info]     └─────────┴────────────┴───────────────────────┘

**Comment by dpositive** [comment]
For now it works. Thanks!
